OpenDreamClient/Interface/DreamInterfaceManager.cs

```diff
@@ -498,34 +505,92 @@ internal sealed class DreamInterfaceManager : IDreamInterfaceManager {
             return true;
         }
 
+        string HandleEmbeddedWinget(string? controlId, string value) {
+            string result = value;
+            int startPos = result.IndexOf("[[");
+            while(startPos > -1){
+                int endPos = result.IndexOf("]]", startPos);
+                if(endPos == -1)
+                    break;
+                string inner = result.Substring(startPos+2, endPos-startPos-2);
+                string[] elementSplit = inner.Split('.');
+                string innerControlId = controlId ?? "";
+                if(elementSplit.Length > 1){
+                    innerControlId = (string.IsNullOrEmpty(innerControlId) ? "" : innerControlId+".")+string.Join(".", elementSplit[..^1]);
+                    inner = elementSplit[^1];
+                }
+                string innerResult = WinGet(innerControlId, inner);
+                result = result.Substring(0, startPos) + innerResult + result.Substring(endPos+2);
+                startPos = result.IndexOf("[[");
+            }
+            return result;
+        }
+
         if (string.IsNullOrEmpty(controlId)) {
             List<DMFWinSet> winSets = parser.GlobalWinSet();
 
             if (CheckParserErrors())
                 return;
 
             // id=abc overrides the elements of other winsets without an element
-            string? elementOverride = winSets.FirstOrNull(winSet => winSet.Element == null && winSet.Attribute == "id")?.Value;
+            string? elementOverride = null;
+            foreach(var winSet in winSets)
+                if(winSet.Attribute == "id" && winSet.Element == null)
+                    elementOverride = HandleEmbeddedWinget(controlId, winSet.Value);
```

Comment:
Why this change? Does having multiple `id=...` do anything?

Reply:
Id doesn't need to be the first element for it to work, and naturally goon has several instances where it isn't

Reply:
`FirstOrNull()` doesn't return the first value, it returns the earliest value that matches the predicate

Reply:
Oh there was also an issue with using FirstOrNull() when DMFWinset was no longer a struct? I don't super remember, I'll check it

Reply:
```
The type 'DMFWinSet' must be a non-nullable value type in order to use it as parameter 'TSource' in the generic type or method 'Extensions.FirstOrNull<TSource>(IEnumerable<TSource>, Func<TSource, bool>)'[CS0453](https://msdn.microsoft.com/query/roslyn.query?appId%3Droslyn%26k%3Dk%28CS0453%29)
```
Seems like you can't do `FirstOrNull()` on classes, only structs. I do not know why.

Reply:
`FirstOrNull()` is an RT extension for use with value types. Looks like `FirstOrDefault()` would be what you want here.